python/cuml/manifold/umap.pyx

```diff
@@ -272,10 +272,72 @@ class UMAP(Base):
 
         self.umap_params = <size_t > umap_params
 
+    def __getstate__(self):
+        state = self.__dict__.copy()
+
+        del state['handle']
+
+        cdef size_t params_t = <size_t>self.umap_params
+        cdef UMAPParams* umap_params = <UMAPParams*>params_t
+
+        state['X_m'] = cudf.DataFrame.from_gpu_matrix(self.X_m)
+        state['arr_embed'] = cudf.DataFrame.from_gpu_matrix(self.arr_embed)
+        state["n_neighbors"] = umap_params.n_neighbors
```

Comment:
This is not a bad idea. I've been trying to think of ways to make the parameters more maintainable within the object. A few params do get adjusted in the C++ layer and I'm thinking it may be best to just pass those parameters as reference and ditch this whole `umap_params` class altogether. It's left over from the stateful variant.